.github/workflows/build.yml

```diff
@@ -140,103 +140,122 @@ jobs:
         with:
           fetch-depth: 0
       - name: Build
-        run: docker build --target release -t $DOCKER_IMAGE_ID .
+        run: |
+          docker buildx create \
+            --name multibuilder \
+            --platform linux/amd64,linux/arm64 \
+            --bootstrap --use
+          docker buildx build \
+            --platform linux/amd64,linux/arm64 \
+            -t $DOCKER_IMAGE_ID .
       - name: Check
         run: |
+            docker buildx build --load -t $DOCKER_IMAGE_ID .
+            # Assert that simple cases works for the new built image
             docker run $DOCKER_IMAGE_ID version
             docker run $DOCKER_IMAGE_ID --help
-            docker run $DOCKER_IMAGE_ID help
-            docker run $DOCKER_IMAGE_ID run --help
-            docker run $DOCKER_IMAGE_ID inspect --help
-            docker run $DOCKER_IMAGE_ID status --help
-            docker run $DOCKER_IMAGE_ID stats --help
-            docker run $DOCKER_IMAGE_ID scale --help
-            docker run $DOCKER_IMAGE_ID pause --help
-            docker run $DOCKER_IMAGE_ID resume --help
 
-      - name: Build browser-enabled image
-        run: docker build -t $DOCKER_IMAGE_ID:with-browser --target with-browser .
       - name: Log into ghcr.io
         if: ${{ github.ref == 'refs/heads/master' || startsWith(github.ref, 'refs/tags/v') }}
         run: |
+          if [[ `git status --porcelain` ]]; then
+            echo "The repository has unexpected changes identified from the git history."\
+            "Force the fail as the job can't guarantee that the code is not changed between the previous build and the next build+push."
+            exit 1
+          fi
           # Log into GitHub Container Registry
           echo "${{ secrets.GITHUB_TOKEN }}" | docker login https://ghcr.io -u ${{ github.actor }} --password-stdin
       - name: Publish k6:master image to ghcr.io
         if: ${{ github.ref == 'refs/heads/master' }}
         run: |
           echo "Publish as ghcr.io/$GHCR_IMAGE_ID:$VERSION"
-          docker tag "$DOCKER_IMAGE_ID" "ghcr.io/$GHCR_IMAGE_ID:master"
-          docker push "ghcr.io/$GHCR_IMAGE_ID:master"
-          docker tag "$DOCKER_IMAGE_ID:with-browser" "ghcr.io/$GHCR_IMAGE_ID:master-with-browser"
-          docker push "ghcr.io/$GHCR_IMAGE_ID:master-with-browser"
+          docker buildx build --push \
+            --platform linux/amd64,linux/arm64 \
+            -t ghcr.io/$GHCR_IMAGE_ID:master .
```

Comment:
I guess this - using `docker buildx build --push` to push the images is the difference with what #3128 reverted ? 

Can we have this tested with some tag and have someone on a macos + arm test it? Or have we done this already?

Reply:
I could split it in a PR adding only ARM64 for master-with-browser, and then they could test it after we've merged. WDYT? We have tested the build not the push.

Reply:
You can also copy one of the sections below and make it push a build for just any commit - that should build and push it when you push to commit and we can test that.

Reply:
Tested, the browser told me that is works